src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryMath.cs

```diff
@@ -193,25 +331,101 @@ public static FormulaValue SumTable(EvalVisitor runner, SymbolContext symbolCont
         // Max(1,2,3)     
         internal static FormulaValue Max(IRContext irContext, FormulaValue[] args)
         {
-            return RunAggregator(new MaxAgg(), irContext, args);
+            if (irContext.ResultType == FormulaType.Number)
+            {
+                return RunAggregator(new MaxNumberAgg(), irContext, args);
+            }
+            else if (irContext.ResultType == FormulaType.DateTime)
+            {
+                return RunAggregator(new MaxDateTimeAgg(), irContext, args);
+            }
+            else if (irContext.ResultType == FormulaType.Date)
+            {
+                return RunAggregator(new MaxDateAgg(), irContext, args);
+            }
+            else if (irContext.ResultType == FormulaType.Time)
+            {
+                return RunAggregator(new MaxTimeAgg(), irContext, args);
+            }
+            else
+            {
+                return new BlankValue(irContext);
+            }
         }
 
         // Max([1,2,3], Value * Value)     
         public static FormulaValue MaxTable(EvalVisitor runner, SymbolContext symbolContext, IRContext irContext, FormulaValue[] args)
         {
-            return RunAggregator(new MaxAgg(), runner, symbolContext, irContext, args);
+            if (irContext.ResultType == FormulaType.Number)
+            {
+                return RunAggregator(new MaxNumberAgg(), runner, symbolContext, irContext, args);
+            }
+            else if (irContext.ResultType == FormulaType.DateTime)
+            {
+                return RunAggregator(new MaxDateTimeAgg(), runner, symbolContext, irContext, args);
+            }
+            else if (irContext.ResultType == FormulaType.Date)
+            {
+                return RunAggregator(new MaxDateAgg(), runner, symbolContext, irContext, args);
+            }
+            else if (irContext.ResultType == FormulaType.Time)
+            {
+                return RunAggregator(new MaxTimeAgg(), runner, symbolContext, irContext, args);
+            }
```

Comment:
Can we do it like this (shares the call to RunAggregator)):

            IAggregator agg;
            if (irContext.ResultType == FormulaType.Number)
            {
                agg = new MaxNumberAgg();
            }
            else if (irContext.ResultType == FormulaType.DateTime)
            {
                agg = new MaxDateTimeAgg(), irContext, args);
            }
            else if (irContext.ResultType == FormulaType.Date)
            {
                agg = new MaxDateAgg();
            }
            else if (irContext.ResultType == FormulaType.Time)
            {
                agg = new MaxTimeAgg();
            }
   else
            {
                return new BlankValue(irContext);
            }
   return RunAggregator(agg, irContext, args);